zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java

```diff
@@ -16,10 +16,9 @@
  */
 package org.apache.zeppelin.socket;
 
-import com.google.common.base.Strings;
-import com.google.gson.Gson;
-import com.google.gson.GsonBuilder;
-import com.google.gson.reflect.TypeToken;
+import shaded.com.google.gson.Gson;
+import shaded.com.google.gson.GsonBuilder;
+import shaded.com.google.gson.reflect.TypeToken;
```

Comment:
You are right. Perhaps I was a bit too hasty.
More generally, I see a problem with the Guava lib, which exists as a transitive dependency in different versions. That's why I opened #4217.

Reply:
We need to use the shaded GSON class at this point because `Input.TypeAdapterFactory` is part of the Zeppelin interpreter.
`Input.TypeAdapterFactory` uses the shaded GSON from the Zeppelin server's point of view.
Without this change you get the following compilation warning.
```
[ERROR] /home/pdallig/git/zeppelin/zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java:[139,40] incompatible types: org.apache.zeppelin.display.RuntimeTypeAdapterFactory cannot be converted to com.google.gson.TypeAdapterFactory
[ERROR] /home/pdallig/git/zeppelin/zeppelin-server/src/main/java/org/apache/zeppelin/socket/ConnectionManager.java:[64,40] incompatible types: org.apache.zeppelin.display.RuntimeTypeAdapterFactory cannot be converted to com.google.gson.TypeAdapterFactory
```